sirhelpers.h

```diff
@@ -181,21 +195,19 @@ bool _sir_validptrnofail(const void* restrict p) {
     return __sir_validptr(p, false);
 }
 
+/** Validates a pointer-to-pointer and optionally fails if it's invalid. */
 bool __sir_validptrptr(const void* restrict* pp, bool fail);
 
 /** Validates a pointer and fails if it's invalid. */
 #define _sir_validptr(p) __sir_validptr((const void* restrict)p, true)
 
-/** Validates a pointer-to-pointer and fails if it's invalid. */
-#define _sir_validptrptr(pp) __sir_validptrptr((const void* restrict*)pp, true)
-
 /** Validates a pointer-to-function and fails if it's invalid. */
 #define _sir_validfnptr(fnp) __sir_validptrptr((const void* restrict*)&fnp, true)
 
 /** Places a null terminator at the first index in a string buffer. */
 static inline
 void _sir_resetstr(char* str) {
-    str[0] = '\0';
+    str[0] = (char)'\0';
```

Comment:
I also removed this on purpose; it's a remnant from when I had a typedef for sirchar, and thought some day I might do wide string support.